samples/snippets/sample_pagination.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START compute_images_list_page ]
+# [START compute_images_list ]
+import google.cloud.compute_v1 as compute_v1
+# [END compute_images_list ]
+# [END compute_images_list_page ]
+
+
+# [START compute_images_list ]
+def print_images_list(project: str) -> None:
+    """
+    Prints a list of all non-deprecated image names available in given project.
+
+    Args:
+        project: project ID or project number of the Cloud project you want to list images from.
+
+    Returns:
+        None
+    """
+    images_client = compute_v1.ImagesClient()
+    # Listing only non-deprecated images to reduce the size of the reply.
+    images_list_request = compute_v1.ListImagesRequest(project=project, max_results=3,
+                                                       filter="deprecated.state != DEPRECATED")
+
+    # Although the max_results parameter is specified as 3 in the request, the iterable returned
+    # by the `list()` method hides the pagination mechanic and allows you to simply iterate over
+    # all the images, while the library makes multiple requests to the API for you.
```

Comment:
Cleaning up the text a little bit:

```suggestion
    # Although the max_results parameter is specified in the request, the iterable returned
    # by the `list()` method hides the pagination mechanic. The library makes multiple
    # requests to the API for you, so you can simply iterate over all the images.
```